webrender/src/device/gfx/device.rs

```diff
@@ -3374,11 +3219,17 @@ impl<B: hal::Backend> Device<B> {
             self.depth_available,
             "Enabling depth test without depth target"
         );
-        self.current_depth_test = Some(LESS_EQUAL_TEST);
+        if mem::replace(&mut self.current_depth_test, Some(LESS_EQUAL_TEST)).is_none() && self.inside_renderpass {
```

Comment:
as we discussed, this shouldn't be needed

Reply:
Fixed it in 88a94948063607aae7dbd735b56b8483118e77b5. The overhead is that we need to acquire depth images for all the render target Textures.

Reply:
We shouldn't allocate depth images for all render targets. Today, if there is any task associated with the target that requires depth, we use a render pass with depth, and of course we'd need to have the depth allocated. But if we pick a render pass without depth, there is no need to have the depth image.

Reply:
In my last commit I've removed render passes without depth completely to work around the issue which I've mentioned on gitter:
> So the problem is that we use a Subpass for creating the GraphicsPipelineDesc and we pick that depending on the depth test : https://github.com/szeged/webrender/blob/master/webrender/src/device/gfx/program.rs#L233 . When we start a render pass we have some kind of depth test state and we pick a sub pass for that. If the depth test changes we pick our pipeline for the draw according to the new depth test state. What's conflicting in this case the sub pass which we picked at begin_render_pass and the sub pass in the GraphicsPipelineDesc of the newly picked pipeline, because they have incompatible attachments.

For this to work we need to have a depth image for all Framebuffer, that's why I came up with the idea to acquire depth images for all the render target Textures. But you're right that we should not allocate depth images for all render targets.
Since we don't want  to write those extra images we could just use a dummy depth image, but with the proper bounds to avoid  ``` vkCreateFramebuffer(): VkFramebufferCreateInfo attachment #1 mip level 0 has dimensions smaller than the corresponding framebuffer dimensions. Here are the respective dimensions for attachment #1, framebuffer:``` validation layer errors. So in my last modification: a377198ab58e8042030565ff7e666c07ced7a4eb I'm creating on of the frame depth image with a sufficient size and use it as a placeholder image in those places.

Reply:
Sorry, I think we are still doing the wrong thing here. This is the problem:
> What's conflicting in this case the sub pass which we picked at begin_render_pass and the sub pass in the GraphicsPipelineDesc

In order to solve it we need 3 pipelines (instead of 2):
  - depth enabled
  - depth disabled in a pass that has depth enabled
  - depth disabled

This would allow us to completely avoid having a depth image.

Reply:
So the solution turned out to be very simple: a6d1466a6bbde77108b215e655a4c3de056408df, without any depth image addition or render pass removal.
This works for now, however I think I should revisit our depth test setters/getters, because I'm not sure we are doing the right thing there.

Reply:
I think we still need different pipelines for render passes with/without depth (pipelines with disabled depth, that is).

Reply:
I've addressed it in my last commit: https://github.com/szeged/webrender/pull/312/commits/92b94329c04e578f1dc51128597f9061b2636fdf .